lib/plurimath/mathml/parser.rb

```diff
@@ -9,17 +9,38 @@ class Parser
       attr_accessor :text
 
       def initialize(text)
-        @text = text.gsub(/\s/, "")
+        @text = text
       end
 
       def parse
-        tree_t = Plurimath::Mathml::Parse.new.parse(text)
-        tree_t = JSON.parse(tree_t.to_json, symbolize_names: true)
-        formula = Plurimath::Mathml::Transform.new.apply(tree_t)
-        formula = [formula] unless formula.is_a?(Array) || formula.nil?
-        return if formula.nil?
+        nodes = JSON.parse(
```

Comment:
This seems to be a performance issue, if you're converting Ox nodes into JSON, and then parsing JSON into a Hash. Can't we directly convert the Ox nodes into a Hash?